src/rgw/rgw_rest_swift.cc

```diff
@@ -1257,6 +1257,21 @@ int RGWDeleteObj_ObjStore_SWIFT::verify_permission()
   }
 }
 
+int RGWDeleteObj_ObjStore_SWIFT::verify_permission(Jager_Tracer& tracer, const Span& parent_span)
+{
+  Span span = tracer.child_span("rgw_rest_swift.cc RGWDeleteObj_ObjStore_SWIFT::verify_permission", parent_span);
```

Comment:
you mentioned you were going to try using a global tracer, can you explain your approach here?

Reply:
@ideepika the tracer is global, but the spans are not, each Root span denotes a new client request and that root spans will contain its serveral follow_up spans and child spans.
In the above example I m creating a new child span, `parent_span` is the span created in the function which is calling this function

Reply:
@suab321321 I understand that, my question is : is it important to pass the tracer to the function, or are you keeping the tracer globally unique for rgw?

Reply:
@ideepika yes it is mandatory to pass the tracer as an argument, even if I make the tracer an `extern` and use this as global tracer, but I wont be able to make the `Span` an extern variable.I understand that by making and `tracer` and `Span` global I dont have to pass the variable as argument which in turn will prevent the creating of overloaded function, but the span variable is an issue here